Add delete button for dag runs in more options menu

dheerajturaga opened the pull request:
- Add 'more options' hamburger menu button next to 'Mark Dag Run as' dropdown in dag run header
- Move delete functionality into the more options menu following the same pattern as dag header
- Update DeleteRunButton styling to match delete dag button theme (remove solid variant, add Box wrapper)
- Add conditional navigation: only navigate to /dags/<dag_id>/runs when deleting from specific run detail page

<img width="1910" height="747" alt="image" src="https://github.com/user-attachments/assets/92434524-fdd0-4c01-a251-0fcde8af8ace" />

<img width="1917" height="737" alt="image" src="https://github.com/user-attachments/assets/6704aa62-ebda-4072-8841-0d92c51df65a" />

dheerajturaga committed:
Add delete button for dag runs in more options menu

  - Add 'more options' hamburger menu button next to 'Mark Dag Run as' dropdown in dag run header
  - Move delete functionality into the more options menu following the same pattern as dag header
  - Update DeleteRunButton styling to match delete dag button theme (remove solid variant, add Box wrapper)
  - Add conditional navigation: only navigate to /dags/<dag_id>/runs when deleting from specific run detail page

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/DeleteRunButton.tsx:
```diff
@@ -28,29 +29,36 @@ import { useDeleteDagRun } from "src/queries/useDeleteDagRun";
 
   return (
-    <>
+    <Box width={width}>
```
I feel like we should set button width via `...props` and set the box width only inside the menu created in Header.tsx

dheerajturaga replied:
Let me know if the current changes are satisfactory

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/Run/Header.tsx:
```diff
@@ -90,6 +92,22 @@ export const Header = ({
+                <Menu.Positioner>
+                  <Menu.Content>
+                    <Menu.Item closeOnSelect={false} value="delete">
+                      <DeleteRunButton dagRun={dagRun} width="100%" withText={true} />
```
Why not put the Box here instead?

dheerajturaga replied:
Done

dheerajturaga committed:
Brent's suggestions

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/DeleteRunButton.tsx:
```diff
@@ -28,17 +29,24 @@ import { useDeleteDagRun } from "src/queries/useDeleteDagRun";
+} & ButtonProps;
 
-const DeleteRunButton = ({ dagRun, withText = true }: DeleteRunButtonProps) => {
+const DeleteRunButton = ({ dagRun, width, withText = true }: DeleteRunButtonProps) => {
```
```suggestion
const DeleteRunButton = ({ dagRun, withText = true, ...rest }: DeleteRunButtonProps) => {
```

dheerajturaga replied:
Done!

bbovenzi commented on airflow-core/src/airflow/ui/src/pages/DeleteRunButton.tsx:
```diff
@@ -50,7 +58,7 @@ const DeleteRunButton = ({ dagRun, withText = true }: DeleteRunButtonProps) => {
         text={translate("dags:runAndTaskActions.delete.button", { type: translate("dagRun_one") })}
-        variant="solid"
+        width={width}
         withText={withText}
```
```suggestion
        withText={withText}
        {...rest}
```

bbovenzi replied:
Can we make sure that the variant is respected over in DagRuns.tsx?

dheerajturaga replied:
Done!

dheerajturaga committed:
Refactor DeleteRunButton to use rest props pattern

  Use rest/spread pattern instead of explicit width prop to allow
  passing any ButtonProps to the underlying ActionButton component

 Fixed TypeScript errors by excluding conflicting onClick and colorPalette props from ButtonProps in DeleteRunButtonProps type definition.

dheerajturaga committed:
ensures that the variant prop can be passed through the rest pattern
